src/api/app/assets/javascripts/webui/notification.js

```diff
@@ -30,3 +30,9 @@ function handleNotificationCheckboxSelection() { // jshint ignore:line
   });
 }
 
+function setSpinnerAndDisableButtons() { // jshint ignore:line
```

Comment:
I believe this function could be replaced with [automatic disabling from Rails unobtrusive JavaScript](https://guides.rubyonrails.org/working_with_javascript_in_rails.html#automatic-disabling). This implies setting on the button the data attribute `data-disable-with: 'Marking as "Read"...'` (or Unread depending on the case)

Reply:
ahh cool, I will have a look

Reply:
@dmarcoux @eduardoj I applied the ujs way, it know disables the button and show a spinner instead of the text, let me know what you think :)